optuna/distributions.py

```diff
@@ -529,9 +529,12 @@ def to_external_repr(self, param_value_in_internal_repr: float) -> CategoricalCh
         return self.choices[int(param_value_in_internal_repr)]
 
     def to_internal_repr(self, param_value_in_external_repr: CategoricalChoiceType) -> float:
-        for index, choice in enumerate(self.choices):
-            if _categorical_choice_equal(param_value_in_external_repr, choice):
-                return index
+        try:
+            return self.choices.index(param_value_in_external_repr)
+        except ValueError:
```

Comment:
Could you add a comment here? It is difficult to recognize why `ValueError` is raised.

Reply:
Thank you for your update https://github.com/optuna/optuna/pull/5400/commits/9e65e7b54c068be9610bc451a7778592f12722a8. But, could you please make clarify more? Normally, `index` is the same as for-if in `except`, but the behavior is different only when the elements contain NaN.

FYI:
```python
(0, 1, float("nan")).index("nan")  # raise ValueError
```

Reply:
Ok, I added another line!